src/Workspaces/Core/Portable/Workspace/Solution/ChecksumCollection.cs

```diff
@@ -58,7 +58,7 @@ public void AddAllTo(HashSet<Checksum> checksums)
         AssetPath assetPath,
         TextDocumentStates<TState> documentStates,
         HashSet<Checksum> searchingChecksumsLeft,
-        Dictionary<Checksum, object> result,
+        Action<Checksum, object> onAssetFound,
```

Comment:
instead of passing in dictionaries, we pass in callbacks to invoke when the item is found.  In this Pr, this will still end up populating a dictionary.  However in https://github.com/dotnet/roslyn/pull/73013 this changes to populating a channel that then allows another task to be reading from the channle when items are found, writing the assets out to our pipe between oop and the host.